opendevin/lib/event.py

```diff
@@ -1,9 +1,14 @@
 import os
 import json
-import agenthub.langchains_agent.utils.actions as actions
+import opendevin.lib.actions as actions
+
+ACTION_TYPES = ['run', 'kill', 'browse', 'read', 'write', 'recall', 'think', 'output', 'error', 'finish']
```

Comment:
Hmm this is a really good point.

I do think we'll want structured actions outside the agent to highlight certain things in the UI (e.g. if we know which file was written, we can change what file is visible in the UI editor, or if we know it's browsing, we can highlight the browser). 

But letting the agent specify the natural language message is also important. Maybe we add a `message` field to `Event`?

Worth noting: with the current setup, not every `Event` is created by the agent. Some Events represent command output, file contents, or HTML from the browser. So not all of them would have a `message`--but we could display the ones that do in the chat log.

Reply:
Good point! We will be needing some metadata for the UI -- If that's the case, maybe we can make this into an Action class with two types: Message or Action (and under action, we can browse, read, write, etc).

But i guess the tricky thing is that those changes would be pretty hard to detect, if the model modifies those file with `bash` instead of using the provided tool. but let's worry about it when it comes up

Reply:
Maybe we can just do 

The input to Agent:
- natural language 

The output of Agent:
- Action: which is Message OR Event

then we have 
- class Message (as is)
- class Event -> that encapsulates most action you have now?